src/hotspot/share/services/mallocHeader.hpp

```diff
@@ -120,9 +120,9 @@ class MallocHeader {
 public:
   // Contains all of the necessary data to to deaccount block with NMT.
   struct FreeInfo {
-    const size_t size;
-    const MEMFLAGS flags;
-    const uint32_t mst_marker;
```

Comment:
I need help with this part. I'm not familiar with C++ enough to make my change work while keeping these fields `const`

Can someone suggest a better approach here?

Reply:
Its a warning sign that something is off. From first glance, a FreeInfo should be immutable; there should be no reason to modify it once its created.